roachtest: add knex test

otan opened the pull request:
These tests *will* fail on master as knex does not handle the PK/FK name
rename correctly. However, this commit alone makes it backportable to
v21.2, so this is being presented as is.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73773)
<!-- Reviewable:end -->

rafiss commented on pkg/cmd/roachtest/tests/knex.go:
```diff
@@ -0,0 +1,130 @@
+		require.NoError(t, err)
+
+		t.Status("running knex tests")
+		// Ignore the error, this is expected to fail.
```
this says ignore the error, but line 117 calls fatal on it. i think in this case since all  tests pass, the `fatal` is correct, so the comment should be updated

rafiss reviewed (approved):
small nit then feel free to merge!

otan committed:
roachtest: add knex test

These tests *will* fail on master as knex does not handle the PK/FK name
rename correctly. However, this commit alone makes it backportable to
v21.2, so this is being presented as is.

Release note: None

otan commented:
bors r=rafiss
